identity/auth-starter/src/main/java/io/camunda/identity/authentication/oidc/OidcCurrentUserService.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.identity.authentication.oidc;
+
+import io.camunda.identity.authentication.CurrentUserService;
+import io.camunda.identity.record.CamundaUser;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+
+@Service
+@Profile("identity-oidc-auth")
+public class OidcCurrentUserService implements CurrentUserService {
+  @Override
+  public CamundaUser getCurrentUser() {
```

Comment:
I'm thinking if we need a separate implementation for oidc, as the authorities can be retrieved from authentication object (it would be an instance of OAuth2AuthenticationToken). The only difference is using authentication.name which extracted from claims["name"] instead of principal.getPreferredUsername which filled from claims["preferred_username"].